File: src/main/java/com/timxyz/services/StatusService.java
package com.timxyz.services;

import com.timxyz.models.Status;
import com.timxyz.repositories.StatusRepositoy;
import com.timxyz.services.exceptions.ServiceException;
import java.util.Collection;
import org.springframework.stereotype.Service;

/**
 * Created by Dario on 5/6/2017.
 */
@Service
public class StatusService extends BaseService<Status, StatusRepositoy> {

    public Status save(Status model) throws ServiceException {
        Status sameName = getByName(model.getName());
         if (sameName != null && model.getId() != sameName.getId()) {
            throw new ServiceException("A category with this name already exists!");
        }
        return super.save(model);
    }

    public Status getByName(String name) {
        return repository.findFirstByName(name);}

    public Collection<Status> filterByName(String name) {
        return repository.filterByName(name);
    }
}

File: frontend/client/app/services/statusService.js
import BaseService from './baseService';

export default class StatusService extends BaseService {
	all() {
		return super.get('status/all');
	}

	create(status) {
		return super.post('status', status);
	}

    getPage(page) {
        return super.get('status/page/' + page);
    }

    delete(id) {
		return super.delete('status/' + id);
	}

    find(id) {
        return super.get('status/' + id);
    }

    update(id, status) {
        return super.post('status/' + id, status);
    }

    filterByName(name) {
        return super.get('status/filter-by/name?name=' + name);
    }
}
